src/PrestaShopBundle/Controller/Admin/ProductController.php

```diff
@@ -1190,8 +1193,11 @@ public function toggleStatusAction($productId)
 
         try {
             $isEnabled = $this->getQueryBus()->handle(new GetProductIsEnabled($productId));
-
-            $this->getCommandBus()->handle(new UpdateProductStatusCommand($productId, !$isEnabled));
+            $this->getCommandBus()->handle(new UpdateProductStatusCommand(
+                $productId,
+                !$isEnabled,
+                ShopConstraint::shop((int) $this->getContextShopId())
```

Comment:
I think this won't work when you are in All shops mode You should the `getShopConstraint()` you introduced here instead